packages/extension-ui/src/Popup/Signing/LedgerSign.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019-2021 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ExtrinsicPayload } from '@polkadot/types/interfaces';
+
+import { faSync } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import React, { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import { Button, Warning } from '../../components';
+import { useLedger } from '../../hooks/useLedger';
+import useTranslation from '../../hooks/useTranslation';
+
+interface Props {
+  accountIndex?: number;
+  addressOffset? : number;
+  className?: string;
+  error: string | null;
+  genesisHash?: string;
+  onSignature?: ({ signature }: { signature: string }) => void;
+  payload?: ExtrinsicPayload;
+  setError: (value: string | null) => void;
+}
+
+function LedgerSign ({ accountIndex, addressOffset, className, error, genesisHash, onSignature, payload, setError } : Props): JSX.Element {
+  const [isBusy, setIsBusy] = useState(false);
+  const { t } = useTranslation();
+  const { error: ledgerError, isLoading: ledgerLoading, isLocked: ledgerLocked, ledger, refresh, warning: ledgerWarning } = useLedger(genesisHash, accountIndex, addressOffset);
+
+  useEffect(() => {
+    if (ledgerError) {
+      setError(ledgerError);
+    }
+  }, [ledgerError, setError]);
+
+  const _onRefresh = useCallback(() => {
+    refresh();
+    setError(null);
+  }, [refresh, setError]);
+
+  const _onSignLedger = useCallback((): void => {
+    if (!ledger || !payload || !onSignature) {
+      return;
+    }
+
+    setError(null);
+    setIsBusy(true);
+    ledger.sign(payload.toU8a(true), Number(accountIndex), Number(addressOffset))
```

Comment:
This seriously worries me. This feels like a plaster on some issue, somewhere.

Reply:
It definitely is. I'll dig a bit. It happens since the first time I've played with the Dropdown. Getting an error for ledger, along the line "only integer are allowed".

Reply:
The string casting comes from the Dropdown, that is an enhanced HTML `select`. The "target.event.value" coming from the `onChange` is a string, this comes from the `HTMLSelectElement`. I've tried to extend this type to support `number` as a value, but failed so far. The best I could do is cast these right away in the onchange, because I know they are of type number.

Reply:
Ahhh. Ok, you may want to do the `Number(...)` in a useCallback then? I'm more worried that it propagates everywhere at this point - UI, back-end, may end up in the JSON, etc.

Reply:
... which is indeed what you did :)